src/Controllers/RegisterController.php

```diff
@@ -57,26 +57,24 @@ public function registerAction(): RedirectResponse
         }
 
         // Save the user
-        $allowedPostFields = array_merge(setting('Auth.validFields'), setting('Auth.personalFields'));
-        $user              = $this->getUserEntity();
-
+        $allowedPostFields = array_merge(
+            setting('Auth.validFields'),
+            setting('Auth.personalFields'),
+            ['password']
+        );
+        $user = $this->getUserEntity();
         $user->fill($this->request->getPost($allowedPostFields));
 
         // Workaround for email only registration/login
         if ($user->username === null) {
             $user->username = null;
         }
 
-        if (! $users->save($user)) {
-            return redirect()->back()->withInput()->with('errors', $users->errors());
-        }
+        $users->saveWithEmailIdentity($user);
```

Comment:
This makes me a little nervous that we rely solely on the Auth validation rules. A developer could provide their own UserModel with separate validation rules that causes the `save()` to fail but we don't catch the errors.

Reply:
Okay, you have a valid point. I'll change to catch.

Reply:
Added ValidationException and catch it.